lib/ldp.js

```diff
@@ -422,9 +422,21 @@ class LDP {
     }
   }
 
+  // delete resource with acl and meta links
   async deleteResource (path) {
+    const linkPath = this.resourceMapper._removeDollarExtension(path)
     try {
-      return await withLock(path, { mustExist: false }, () => promisify(fs.unlink)(path))
+      // first delete file, then links with write permission only (atomic delete)
+      await withLock(path, { mustExist: false }, () => promisify(fs.unlink)(path))
+
+      const aclPath = `${linkPath}${this.suffixAcl}`
```

Comment:
I would actually expect a listing of _all_ files in the folder, and those should be deleted. As per the above check, they should all be `.meta` and `.acl` files. Although, since we didn't lock from that point to here, we don't know that 100% certain, so better to delete all.

Reply:
This is already done in `async deleteContainer(directory)`, where `rimraf` is used and deletes all files which are expected to only be `.acl` or `.meta` per the above the check.